sdk/v2/core/jobs.go

```diff
@@ -0,0 +1,381 @@
+	JobPhaseTimedOut JobPhase = "TIMED_OUT"
+	// JobPhaseUnknown represents the state wherein a Job's state is unknown. Note
+	// that this is possible if and only if the underlying Job execution substrate
+	// (Kubernetes), for some unanticipated, reason does not know the Job's
+	// (Pod's) state.
+	JobPhaseUnknown WorkerPhase = "UNKNOWN"
+)
+
+// Job represents a component spawned by a Worker to complete a single task
+// in the course of handling an Event.
+type Job struct {
+	// Spec is the technical blueprint for the Job.
+	Spec JobSpec `json:"spec"`
+	// Status contains details of the Job's current state.
+	Status *JobStatus `json:"status"`
+}
+
+// JobSpec is the technical blueprint for a Job.
+type JobSpec struct {
+	// PrimaryContainer specifies the details of an OCI container that forms the
+	// cornerstone of the Job. Job success or failure is tied to completion and
+	// exit code of this container.
+	PrimaryContainer JobContainerSpec `json:"primaryContainer"`
+	// SidecarContainers specifies the details of supplemental, "sidecar"
+	// containers. Their completion and exit code do not directly impact Job
+	// status. Brigade does not understand dependencies between a Job's multiple
+	// containers and cannot enforce any specific startup or shutdown order. When
+	// such dependencies exist (for instance, a primary container than cannot
+	// proceed with a suite of tests until a database is launched and READY in a
+	// sidecar container), then logic within those containers must account for
+	// these constraints.
+	SidecarContainers map[string]JobContainerSpec `json:"sidecarContainers,omitempty"` // nolint: lll
+	// TimeoutSeconds specifies the time, in seconds, that must elapse before a
+	// running Job should be considered to have timed out.
+	TimeoutSeconds int64 `json:"timeoutSeconds,omitempty"`
+	// Host specifies criteria for selecting a suitable host (substrate node) for
+	// the Job. This is useful in cases where a Job requires a specific,
+	// non-default operating system (i.e. Windows) or specific hardware (e.g. a
+	// GPU.)
+	Host *JobHost `json:"host,omitempty"`
+}
+
+// MarshalJSON amends JobSpec instances with type metadata so that clients do
+// not need to be concerned with the tedium of doing so.
+func (j JobSpec) MarshalJSON() ([]byte, error) {
+	type Alias JobSpec
+	return json.Marshal(
+		struct {
+			meta.TypeMeta `json:",inline"`
+			Alias         `json:",inline"`
+		}{
+			TypeMeta: meta.TypeMeta{
+				APIVersion: meta.APIVersion,
+				Kind:       "JobSpec",
+			},
+			Alias: (Alias)(j),
+		},
+	)
+}
+
+// JobContainerSpec amends the ContainerSpec type with additional Job-specific
+// fields.
+type JobContainerSpec struct {
+	// ContainerSpec encapsulates generic specifications for an OCI container.
+	ContainerSpec `json:",inline"`
+	// UseWorkspace indicates whether the Job requires the Worker's shared
+	// workspace (if one exists) to be mounted into the OCI container.
+	UseWorkspace bool `json:"useWorkspace"`
```

Comment:
Yes... that _is_ the shared storage. I renamed it to "workspace," because it felt cleaner and (not that we're a CI system), but "workspace" is familiar terminology for people coming from some other systems, like Jenkins, for instance.

Note one big difference between this and 1.x shared storage is that this storage is not created by the worker, but created _before_ the worker is started. The result is that the worker has access to the workspace (shared storage) which is something that's been requested a few times.